filters/auth/oidc.go

```diff
@@ -327,7 +332,18 @@ func (f *tokenOidcFilter) doOauthRedirect(ctx filters.FilterContext) {
 		return
 	}
 
-	oauth2URL := f.config.AuthCodeURL(fmt.Sprintf("%x", stateEnc), f.authCodeOptions...)
+	opts := f.authCodeOptions
+	if f.queryParams != nil {
+		opts = make([]oauth2.AuthCodeOption, 0, len(f.authCodeOptions)+len(f.queryParams))
+		copy(opts, f.authCodeOptions)
```

Comment:
The `copy` is tricky, it copies [`min(len(src), len(dst))`](https://golang.org/pkg/builtin/#copy) so we need to init `opts` to the length of src which is `f.authCodeOptions`
```go
opts = make([]oauth2.AuthCodeOption, len(f.authCodeOptions), len(f.authCodeOptions)+len(f.queryParams))
```

See https://play.golang.org/p/-c9w6AxpN0-

Reply:
Yeah, that is a good solution. Thank you. Is there anything missing now?

Reply:
code lgtm, a test would be nice :)

Reply:
@rraraujo Just to be clear 
```go
opts = make([]oauth2.AuthCodeOption, 0, len(f.authCodeOptions)+len(f.queryParams))
```
should be changed to
```go
opts = make([]oauth2.AuthCodeOption, len(f.authCodeOptions), len(f.authCodeOptions)+len(f.queryParams))
```